lib/Sema/CSDiagnostics.h

```diff
@@ -1794,6 +1744,132 @@ class FunctionArgApplyInfo {
   }
 };
 
+/// Diagnose a situation there is a mismatch between argument and parameter
+/// types e.g.:
+///
+/// ```swift
+/// func foo(_: String) {}
+/// func bar(_ v: Int) { foo(v) } // `Int` is not convertible to `String`
+/// ```
+class ArgumentMismatchFailure : public ContextualFailure {
```

Comment:
Could you please move `FunctionArgApplyInfo` to the top of the header instead of moving diagnostics which might depend on it?

Reply:
@xedin Sure thing! If possible, I'd still like to keep it below `FailureDiagnostic`, as IMO that should be the first thing you see on opening the header. Does that sound good?

Reply:
Yeah, that sounds good!